hooks/useDragAndDrop.ts

```diff
@@ -0,0 +1,114 @@
+import { useDropzone, FileRejection } from "react-dropzone";
+import { useCallback } from "react";
+import { toast } from "sonner";
+
+type UseDragAndDropConfig = {
+  onDrop: (files: File[]) => void;
+  maxFiles?: number;
+  maxSizeMB?: number;
+  allowedTypes?: string[];
+  disabled?: boolean;
+};
+
+/**
+ * Hook to handle drag-and-drop file upload using react-dropzone
+ * Provides drag state and handlers for file upload
+ */
+export function useDragAndDrop({
+  onDrop,
+  maxFiles = 100,
+  maxSizeMB = 100,
+  allowedTypes = [],
+  disabled = false,
+}: UseDragAndDropConfig) {
+  const maxSizeBytes = maxSizeMB * 1024 * 1024;
+
+  /**
+   * Validate file extension against blocked list
+   */
```

Comment:
Human Readable Code
- actual: comments do not add any additional information not already described in the function name. 
- required: remove redundant dev comments. Code should be human readable.

```suggestion
```